doc/developers/develop.rst

```diff
@@ -520,9 +520,7 @@ _skip_test (default=False)
 _xfail_checks (default=False)
     dictionary ``{check_name: reason}`` of common checks that will be marked
     as `XFAIL` for pytest, when using
-    :func:`~sklearn.utils.estimator_checks.parametrize_with_checks`. This tag
-    currently has no effect on
-    :func:`~sklearn.utils.estimator_checks.check_estimator`.
+    :func:`~sklearn.utils.estimator_checks.parametrize_with_checks`.
```

Comment:
what does it do with `check_estimator` now?
It should also be documented that it doesn't work with classes.

Reading the checks logic, it seems that there are lots of implicit things going on, and I think it would help to clarify them both via the docs and via comments in the code

Reply:
It works with classes if it can be constructed, which is true true for `estimator_checks` in general.